Improve How to determine if a file is an assembly (#25329)
dotnet/docs: docs/standard/assembly/identify.md

3. If **ILDASM** reports that the file is not a portable executable (PE) file, then it is not an assembly. For more information, see the topic [How to: View assembly contents](view-contents.md).  
  
## How to programmatically determine if a file is an assembly  

### Using the AssemblyName class

1. Call the <xref:System.Reflection.AssemblyName.GetAssemblyName%2A?displayProperty=nameWithType> method, passing the full file path and name of the file you are testing.  
  
2. If a <xref:System.BadImageFormatException> exception is thrown, the file is not an assembly.  

This example tests a DLL to see if it is an assembly.  

[!code-csharp[](snippets/identify/csharp/ExampleAssemblyName.cs#AssemblyName)]

[!code-vb[](snippets/identify/visual-basic/ExampleAssemblyName.vb#AssemblyName)]

The <xref:System.Reflection.AssemblyName.GetAssemblyName%2A> method loads the test file, and then releases it once the information is read.  

### Using the PEReader class

1. Install the [System.Reflection.Metadata](https://www.nuget.org/packages/System.Reflection.Metadata/) NuGet package.

> 2. Create a <xref:System.IO.FileStream?displayProperty=nameWithType> instance to read data from the file you are testing.

Reviewer: ```suggestion
2. Create a <xref:System.IO.FileStream?displayProperty=nameWithType> instance to read data from the file you're testing.
```

3. Create a <xref:System.Reflection.PortableExecutable.PEReader?displayProperty=nameWithType> instance, passing your file stream into the constructor.

4. Check the value of the <xref:System.Reflection.PortableExecutable.PEReader.HasMetadata> property. If the value is `false`, the file is not an assembly.

> 5. Call the <xref:System.Reflection.Metadata.PEReaderExtensions.GetMetadataReader%2A> method on your PE reader instance to create a metadata reader.

Reviewer: ```suggestion
5. Call the <xref:System.Reflection.Metadata.PEReaderExtensions.GetMetadataReader%2A> method on the PE reader instance to create a metadata reader.
```
